modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/EitherOfValue.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.value;
+
+import static java.lang.String.format;
+
+import org.mule.runtime.extension.api.exception.IllegalModelDefinitionException;
+import org.mule.sdk.api.annotation.values.OfValues;
+
+/**
+ * Encapsulates either a {@link OfValues} annotation or a {@link org.mule.runtime.extension.api.annotation.values.OfValues}
+ * annotation in order to be able to use a single object to represent both.
+ * <p>
+ * If the {@link EitherOfValue} is used with an annotation other than {@link OfValues} or a
+ * {@link org.mule.runtime.extension.api.annotation.values.OfValues}, an {@link IllegalModelDefinitionException} exception is
+ * thrown.
+ *
+ * @since 4.4.0
+ */
+public class EitherOfValue {
```

Comment:
Replace with an adapter class

Reply:
OfValue is an annotation, how can I use an adapter here?